src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SupplementaryDatasetService.java

```diff
@@ -37,9 +37,9 @@ public void addSupplementaryDatasetEntity(SupplementaryDatasetDTO supplementaryD
           supplementaryDatasetDTO.getPeriodId());
     }
     try {
-      if (existsByExerciseFK(collectionExercise.getExercisePK())) {
+      if (collectionExercise.getSupplementaryDatasetEntity() != null) {
```

Comment:
!=null do you need that?

Reply:
Yeah, if removed it just gets the `SupplementaryDatasetEntity` object and is expecting a `boolean` which java is not happy to deal with in an if

Reply:
Conceptually it's valid that the collection exercise does not have a mapped supplementary dataset entity in the database. As such, if we have mapped these objects so that its null if one doesn't exist, then this condition would be correct.

Reply:
Oh I see sorry, you're both having a Python discussion!